docs/stability-tiers.md

```diff
@@ -0,0 +1,194 @@
+provided here for phasing out unmaintained features but it should be clear under
+what circumstances work "can be avoided" for each tier.
+
+#### Tier 3 - Not Production Ready
+
+The general idea behind Tier 3 is that this is the baseline for inclusion of
+code into the Wasmtime project. This is not intended to be a catch-all "if a
+patch is sent it will be merged" tier. Instead the goal of this tier is to
+outline what is expected of contributors adding new features to Wasmtime which
+might be experimental at the time of addition. This is intentionally not a
+relaxed tier of restrictions but already implies a significant commitment of
+effort to a feature being included within Wasmtime.
+
+Tier 3 features include:
+
+* Inclusion of a feature does not impose unnecessary maintenance overhead on
+  other components/features. Some examples of additions to Wasmtime which would
+  not be accepted are:
+
+  * An experimental feature doubles the time of CI for all PRs.
+  * A change which makes it significantly more difficult to architecturally
+    change Wasmtime's internal implementation.
+  * A change which makes building Wasmtime more difficult for unrelated
+    developers.
+
+  In general Tier 3 features are off-by-default at compile time.
+
+* New features of Wasmtime cannot have major known bugs at the time of
+  inclusion. Landing a feature in Wasmtime requires the feature to be correct
+  and bug-free as best can be evaluated at the time of inclusion. Inevitably
+  bugs will be found and that's ok, but anything identified during review must
+  be addressed.
+
+* Code included into the Wasmtime project must be of an acceptable level of
+  quality relative to the rest of the code in Wasmtime.
+
+* There must be a path to a feature being finished at the time of inclusion.
+  Adding a new backend to Cranelift for example is a significant undertaking
+  which may not be able to be done in a single PR. Partial implementations of a
+  feature are acceptable so long as there's a clear path forward and schedule
+  for completing the feature.
+
+* New components in Wasmtime must have a clearly identified owner who is willing
+  to be "on the hook" for review, updates to the internals of Wasmtime, etc. For
+  example a new backend in Cranelift would need to have a maintainer who is
+  willing to respond to changes in Cranelift's interfaces and the needs of
+  Wasmtime.
+
+This baseline level of support notably does not require any degree of testing,
+fuzzing, or verification. As a result components classified as Tier 3 are
+generally not production-ready as they have not been battle-tested much.
+
+Features classified as Tier 3 may be turned off or removed from Wasmtime as
+well. If a Tier 3 feature is preventing development of other features then the
+owner will be notified. If no response is heard from within a week then the
+feature will be turned off. If no further response happens for a month then
+the feature may be removed from Wasmtime.
```

Comment:
The distinction between "turned off" and "removed" isn't clear in this paragraph. I think you mean that, first, failing CI/fuzz tests will be disabled, and then if the issue still isn't addressed we'll merge a commit removing the implementation entirely. (Although of course that can be reverted later if somebody commits to maintaining it again.) Do I have that right?

Reply:
Indeed! I'll try to clarify here.